services-custom/dynamodb-enhanced/src/main/java/software/amazon/awssdk/enhanced/dynamodb/document/EnhancedDocument.java

```diff
@@ -0,0 +1,533 @@
+
+/**
+ * Interface representing Document API for DynamoDB. Document API operations are used to carry open content i.e. data with no
+ * fixed schema, data that can't be modeled using rigid types, or data that has a schema. This interface specifies all the
+ * methods to access a Document, also provides constructor methods for instantiating Document that can be used to read and write
+ * to DynamoDB using EnhancedDynamoDB client.
+ *
+ * TODO : Add some examples in the Java Doc after API Surface review.
+ */
+@SdkPublicApi
+public interface EnhancedDocument {
+
+    /**
+     * Convenience factory method - instantiates an <code>EnhancedDocument</code> from the given JSON String.
+     *
+     * @param json The JSON string representation of DynamoDB Item.
+     * @return A new instance of EnhancedDocument.
+     */
+    static EnhancedDocument fromJson(String json) {
+        if (json == null) {
+            return null;
+        }
+        return DefaultEnhancedDocument.builder()
+                                      .json(json)
+                                      .attributeConverterProviders(DefaultAttributeConverterProvider.create())
+                                      .build();
+    }
+
+    /**
+     * Convenience factory method - instantiates an <code>EnhancedDocument</code> from the given AttributeValueMap.
+     * @param attributeValueMap - Map with Attributes as String keys and AttributeValue as Value.
+     * @return A new instance of EnhancedDocument.
+     */
+    static EnhancedDocument fromAttributeValueMap(Map<String, AttributeValue> attributeValueMap) {
+        if (attributeValueMap == null) {
+            return null;
+        }
+        return DefaultEnhancedDocument.builder()
+                                      .attributeValueMap(attributeValueMap)
+                                      .attributeConverterProviders(DefaultAttributeConverterProvider.create())
+                                      .build();
+    }
+
+    /**
+     * Convenience factory method - instantiates an <code>EnhancedDocument</code> from the given Map
+     *
+     * @param attributes Map of item attributes where each attribute should be a simple Java type, not DynamoDB type.
+     * @return A new instance of EnhancedDocument.
+     */
+    static EnhancedDocument fromMap(Map<String, Object> attributes) {
+        if (attributes == null) {
+            return null;
+        }
+        DefaultEnhancedDocument.DefaultBuilder defaultBuilder = DefaultEnhancedDocument.builder();
+        attributes.entrySet().forEach(key -> defaultBuilder.add(key.getKey(), key.getValue()));
+        return defaultBuilder.addAttributeConverterProvider(DefaultAttributeConverterProvider.create())
+                             .build();
+    }
+
+    /**
+     * Creates a default builder for {@link EnhancedDocument}.
+     */
+    static Builder builder() {
+        return DefaultEnhancedDocument.builder();
```

Comment:
Do we include the DefaultAttributeConverters here?

Reply:
No, they're in the DefaultEnhancedDocument ctor